doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -31,6 +31,44 @@ There are multiple notions of :gdef:`equality` in Coq:
   other advantages.  ":term:`Convertible <convertible>`" is another way of saying that
   two terms are definitionally equal.
 
+Equality
+--------
+
+.. tacn:: f_equal
+
+   For a goal with the form :n:`f a__1 ... a__n = g b__1 ... b__n`, creates
+   subgoals :n:`f = g` and :n:`a__i = b__i` for the `n` arguments. Subgoals
+   that can be proven by :tacn:`reflexivity` or :tacn:`congruence` are solved
+   automatically.
+
+.. tacn:: reflexivity
+
+   For a goal with the form `t=u`, verifies that `t` and `u` are :term:`convertible`,
+   and if so,
+   solves the goal.  If not, it fails.  It is equivalent to ``apply refl_equal``.
+
+   .. exn:: The conclusion is not a substitutive equation.
```

Comment:
It looks to me that this is an outdated error message (only place where it appears in the code is here). Now, `reflexivity` raises "Cannot find a relation to rewrite." when called on a goal which is not an equality. Since this error message already appears below, no need to repeat it here, but we could expand the sentence "If not, it fails" to explain the two cases of failures (this one and the "Unable to unify" one).

BTW, there is a third possible error messages for this three tactics:

- Tactic failure:  The relation ... is not a declared reflexive relation. Maybe you need to require the Coq.Classes.RelationClasses library.
- Tactic failure:  The relation ... is not a declared symmetric relation. Maybe you need to require the Coq.Classes.RelationClasses library.
- Tactic failure:  The relation ... is not a declared transitive relation. Maybe you need to require the Coq.Classes.RelationClasses library.

Reply:
> only place where [this error message] appears in the code is here

I don't see anywhere in the source, so I will remove it.

> we could expand the sentence "If not, it fails" to explain the two cases of failures (this one and the "Unable to unify" one).

Seems like overkill to enumerate the possible failures.  The error messages seem self-explanitory to me. 

I added the "is not a declared" messages

Reply:
> > only place where [this error message] appears in the code is here
> 
> I don't see anywhere in the source, so I will remove it.

Indeed, that's what I meant.